From ac99dbdfbaa38c3d6fa7501c38bf8fcd63a9b1fb Mon Sep 17 00:00:00 2001
From: rikardo <rikard.olander@hostmobility.com>
Date: Tue, 3 Oct 2023 05:35:55 +0000
Subject: [PATCH 1/1] gpio-pca953x: fix resume from suspend 
  Return all failures as ret variable in resume. 
  Remove else return 0 if there is no regulator
  in devicetree. The reason is that vcc-supply is an optional parameter and if
  resume returns before reactivate regmap, 
  the driver will lose the IO functionality and will need to be reloaded.

---
 drivers/gpio/gpio-pca953x.c | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/drivers/gpio/gpio-pca953x.c b/drivers/gpio/gpio-pca953x.c
index 1b7f050b4f29..ee55d0b9e455 100644
--- a/drivers/gpio/gpio-pca953x.c
+++ b/drivers/gpio/gpio-pca953x.c
@@ -1213,10 +1213,8 @@ static int pca953x_resume(struct device *dev)
 			ret = regulator_enable(chip->regulator);
 			if (ret) {
 				dev_err(dev, "Failed to enable regulator: %d\n", ret);
-				return 0;
+				return ret;
 			}
-		} else {
-			return 0;
 		}
 	}
 
@@ -1231,7 +1229,7 @@ static int pca953x_resume(struct device *dev)
 		return ret;
 	}
 
-	return 0;
+	return ret;
 }
 #endif
 
@@ -1292,9 +1290,7 @@ static SIMPLE_DEV_PM_OPS(pca953x_pm_ops, pca953x_suspend, pca953x_resume);
 static struct i2c_driver pca953x_driver = {
 	.driver = {
 		.name	= "pca953x",
-#if 0
 		.pm	= &pca953x_pm_ops,
-#endif
 		.of_match_table = pca953x_dt_ids,
 		.acpi_match_table = pca953x_acpi_ids,
 	},
-- 
2.34.1

